### PR TITLE
fix: hide vid/aud icons based on permissions

### PR DESCRIPTION
**Pull request by ashwins93**

### CODE REVIEW

**Comment by raviteja83** [comment]
@ashwins93 could you put some screenshots.

File: src/views/footerView.jsx [hunk 1]
```diff
@@ -139,32 +141,36 @@ export const ConferenceFooter = ({ isChatOpen, toggleChat }) => {
   const leftComponents = [<SettingsView key={0} />];
 
   if (!isMobileDevice()) {
+    leftComponents.push(<VerticalDivider key={1} />);
+    if (localPeerRole?.publishParams.allowed?.includes("screen")) {
```
Comment (triptu) [inline src/views/footerView.jsx]: selectIsAllowedToPublish could be used.

**Comment by ashwins93** [comment]
@raviteja83 

As a host 

image

As a guest (no screenshare)

image

As a viewer (no publish rights)

image

File: src/views/footerView.jsx (hunk 1 above)
Comment (ashwins93) [reply src/views/footerView.jsx]: Done

**Comment by triptu** [comment]
is it possible for the settings and message icon to be together on no screenshare permission?
also if only one of audio and video is allowed are the icons centered?

File: src/views/footerView.jsx [hunk 2]
```diff
@@ -79,13 +81,15 @@ export const ConferenceFooter = ({ isChatOpen, toggleChat }) => {
...
   const pluginRef = useRef(null);
+  const isAllowedToPublish = useHMSStore(selectIsAllowedToPublish);
 
   const initialModalProps = {
     show: false,
     title: "",
     body: "",
   };
   const [errorModal, setErrorModal] = useState(initialModalProps);
+  const localPeerRole = useHMSStore(selectLocalPeerRole);
```
Comment (triptu) [inline src/views/footerView.jsx]: doubt: why is this still needed?

Comment (ashwins93) [reply src/views/footerView.jsx]: It's not. Removed.

**Comment by ashwins93** [comment]
> is it possible for the settings and message icon to be together on no screenshare permission?
> also if only one of audio and video is allowed are the icons centered?

Yes. Refer here for screenshot https://github.com/100mslive/hms-video-react/pull/288

**Review by triptu — APPROVED** [review]

**Pull request merged**